upstairs/src/lib.rs

```diff
@@ -9029,6 +9029,13 @@ pub struct Guest {
      */
     bw_tokens: Mutex<usize>, // bytes
     bw_limit: Option<usize>, // bytes per second
+
+    /// Local cache for block size
+    ///
+    /// This is 0 when unpopulated, and non-zero otherwise; storing it locally
+    /// saves a round-trip through the `reqs` queue, and using an atomic means
```

Comment:
Is it eaiser/better to have 0 be a sentinel value vs. `Option`?
I'm not sure if the `Option` has the same ergonomics benefit.

Reply:
You're correct that an `Option` would be more ergonomic / type-safe here.  However, we're relying on atomic operations to make this callable from `&self` functions, and `Option` functions aren't atomic.